pennylane/beta/tapes/tape.py

```diff
@@ -331,6 +334,8 @@ def _process_queue(self):
                 # attempt to infer the output dimension
                 if obj.return_type is qml.operation.Probability:
                     self._output_dim += 2 ** len(obj.wires)
+                elif obj.return_type is qml.operation.State:
+                    self._state_measurement = obj
```

Comment:
Can this not be included within `self._measurements`? It feels strange to have a separate queue for a single object.

Reply:
This has been changed back to a simple flag `_returns_state`. I'm not sure if we can get around having the flag though, since the `jacobian()` method needs to know as well as the QNode `construct()` method (for the complex dtype), so these things I don't think can be delayed to the device side.

Reply:
Update - have now removed `_returns_state` too - now the `jacobian()` just checks directly the return types in `self.measurements`. And QNode `construct()` has easy access to the measurement processes already, so we just check them for the dtype.